charts/nexus3/templates/ingress.yaml

```
@@ -37,8 +37,10 @@ spec:
               servicePort: http
             {{- end }}
   {{- end }}
-  {{- range .Values.service.additionalPorts }}
-    - host: {{ .host | quote }}
+  {{- range additionalPort := .Values.service.additionalPorts }}
+  {{- if $additionalPort.hosts }}
+  {{- range $additionalPort.hosts }}
+    - host: {{ . | quote }}
```

```suggestion
  {{- range additionalPort := .Values.service.additionalPorts }}
  {{- range ternary $additionalPort.hosts (list $additionalPort.host) $additionalPort.hosts }}
    - host: {{ . | quote }}
```
You should then be able to remove the `else` clause and not have to duplicate the code. The syntax might need tweaking depending on the evaluation pattern, but the structure should be right.